crates/flox/src/commands/channel.rs

```diff
@@ -76,122 +80,118 @@ pub struct Search {
+    let channels = flox
+        .channels
+        .iter()
+        .filter_map(|entry| {
+            if HIDDEN_CHANNELS.contains_key(&*entry.from.id) {
+                None
+            } else if DEFAULT_CHANNELS.contains_key(&*entry.from.id) {
+                Some((ChannelType::Flox, entry))
+            } else {
+                Some((ChannelType::User, entry))
+            }
+        })
+        .sorted();
+
+    // Create `registry` parameter for `pkgdb`
+    let mut priority: Vec<String> = Vec::new();
+    let mut inputs = HashMap::new();
+    for (_, entry) in channels {
+        priority.push(entry.from.id.to_string());
+        let input = RegistryInput {
+            from: entry.to.clone(),
+            // TODO: handle `subtrees` and `stabilities`
+            subtrees: None,
+            stabilities: None,
+        };
+        inputs.insert(entry.from.id.to_string(), input);
+    }
+    let registry = Registry {
+        inputs,
+        priority,
+        defaults: RegistryDefaults::default(),
+    };
+
+    // We've already checked that the search term is Some(_)
+    let query = Query::from_str(search_term)?;
+
+    Ok(SearchParams {
+        registry,
+        query,
+        ..SearchParams::default()
+    })
+}
+
+// This is likely to change significantly after the output format of search results is specced out
+fn render_search_results(search_results: &SearchResults) -> Result<()> {
+    let summarized_results = search_results
+        .results
+        .iter()
+        .map(|r| {
+            let path_components = r.attr_path.clone();
+            let flake_attr = if !DEFAULT_CHANNELS.contains_key(r.input.as_str()) {
+                vec![r.input.clone(), path_components.join_with(".").to_string()]
+                    .join_with("#")
+                    .to_string()
+            } else {
+                path_components.join_with(".").to_string()
+            };
+            let d = if let Some(ref d) = r.description {
+                // Some package descriptions contain newline characters
+                // which breaks formatting
+                d.replace('\n', " ")
+            } else {
+                "<no description provided>".into()
+            };
+            (flake_attr, d)
+        })
+        .collect::<Vec<_>>();
+    let attr_col_width = summarized_results
+        .iter()
+        .fold(0, |cw, (attr, _)| usize::max(cw, attr.len()));
+    // Depending on the search query there could be a ton of results, better to
+    // do buffered writes than to lock `stdout` on every write.
+    let mut writer = BufWriter::new(std::io::stdout());
+    for (attr, desc) in summarized_results.iter() {
+        writeln!(&mut writer, "{attr:<attr_col_width$}  {desc}")?;
+    }
+    Ok(())
+}
+
+fn render_search_results_json(search_results: &SearchResults) -> Result<()> {
+    let json = serde_json::to_string(&search_results.results)?;
+    println!("{}", json);
```

Comment:
note: I find a method this small makes code less readable, because what's actually being done is hidden